src/util/construct-graph.ts

```diff
@@ -0,0 +1,120 @@
+import { Dependency } from "@bc-cr/domain/dependencies";
+import { Graph, Visited } from "@bc-cr/domain/graph";
+
+export function constructGraph(
+  dependencies: Dependency[],
+) {
+  /**
+   * restructure dependencies into a map (directed adjacency list graph):
+   * key = project name
+   * value = array of project names which it depends on
+   */ 
+  const initializedGraph = dependencies.reduce((ig: Graph, current) => {
+    initializeVertex(current, ig);
+    return ig;
+  }, {});
+
+  const graph = dependencies.reduce((g: Graph, current) => {
```

Comment:
why `g` instead of `graph` or `previous` since it is inside a `reduce` block?
why `current` is not typed?